controllers/api/v1/favorites_controller.js

```diff
@@ -48,10 +48,28 @@ async function index(req, res) {
       }
       res.status(200).send(FavoritesSerializer.parse(favorites));
     })
+  } else {
+    res.status(401).send({message: `Invalid credentials.`});
+  }
+}
+
+async function destroy(req, res) {
+  let user = await User.findOne({where: {apiKey:req.body.apiKey || null }});
+  if (user) {
+    let [city, state] = req.body.location.split(',');
+    state = state.trim().toLowerCase();
+    city = city.trim().toLowerCase();
+
+    let location = await Location.findOne({where: {city: city, state: state}});
+    await user.removeLocations(location);
+    res.status(204).send();
+  } else {
+    res.status(401).send({message: `Invalid credentials.`});
   }
```

Comment:
I think this is about as trim as you can get to complete the required functionality :) One thing I did see, I'm curious about the  || null option when searching for a User. Is there something specific you're hoping to catch with that or operator? If not, that syntax is not necessary to achieve the desired result, and I'm not sure if it could open it up for null api_key submissions?

Reply:
Great question! So I was running into issues where is `undefined` is passed into the sequelize method it gave an error so I added the or operator to change the value to null in the case where the API key is not given. Where you able to get it to work with no API key passed in?